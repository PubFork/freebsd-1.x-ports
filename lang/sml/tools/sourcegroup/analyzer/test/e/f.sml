
signature A = sig type t val f :unit-> t end
structure Y:A = struct type t = int * int fun f()=(3,4) end

signature B = sig structure Q:A val x:Q.t val y:int end

signature C = sig val w:int end
structure CC :C = struct val w = 4 end

functor F (E:C) (D:A) :B = struct structure Q:A=D val x:Q.t=Q.f() val y=E.w end

funsig FS (AP:A) = B

functor AA :FS = F (CC)

structure AF :B = AA (Y)

structure XX = F (CC)
