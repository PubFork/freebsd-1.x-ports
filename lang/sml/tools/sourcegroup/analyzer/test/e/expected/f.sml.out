source sml "e/f.sml"
  export structure AF CC XX Y 
  export signature A B C 
  export functor AA F 
  export fsig FS ;

